src/backend/packages/inensus/steama-meter/src/Observers/GeographicalInformationObserver.php

```diff
@@ -31,17 +33,39 @@ public function __construct(
     }
 
     public function updated(GeographicalInformation $geographicalInformation) {
-        if ($geographicalInformation->owner_type === 'meter_parameter') {
-            $meter = $this->meter->newQuery()->find($geographicalInformation->owner_id);
-            $stmMeter = $this->stmMeter->newQuery()->where('mpm_meter_id', $meter->id)->first();
-            if ($stmMeter) {
-                $points = explode(',', $geographicalInformation->points);
-                $putParams = [
-                    'latitude' => floatval($points[0]),
-                    'longitude' => floatval($points[1]),
-                ];
-                $this->stmMeterService->updateSteamaMeterInfo($stmMeter, $putParams);
+        if ($geographicalInformation->owner_type === 'address') {
+            $address = Address::find($geographicalInformation->owner_id);
```

Comment:
One tiny change: we can access `$address`  directly from `$geographicalInformation->owner`